Foody/Onboarding/Register/RegisterViewController.swift

```diff
@@ -6,24 +6,161 @@
+        let twiceDate = calender.startOfDay(for: Date())
+        let companents = calender.dateComponents([.year], from: firstDate, to: twiceDate)
+        
+        if companents.year! >= 15 {
+            
+            makeAlert(title: "Registration Successful.", message:"You are welcome. :)" )
+            
+            //Navigation
+            
+        }else {
+            makeAlert(title: "Warning!", message: "Persons under the age of 15 cannot enter.")
+        }
+        return companents.year!
+    }
+    
+    @IBAction func signUpClicked(_ sender: Any) {
+        
+        // email and password validation
+        
+        if let username = usernameText.text , let email = emailText.text , let password = passwordText.text {
+            if username == "" && email == "" && password == "" {
+                
+                makeAlert(title: "Warning!", message: "Please Enter Username, Email and Password!" )
+            }else {
+                if !email.isValidEmail(email: email) {
+                    makeAlert(title: "Warning!", message: "Please Enter Valid Email!")
+                    
+                }else if !password.isValidPassword(password: password){
+                    makeAlert(title: "Warning!", message: "Please Enter Valid Password!")
+                    
+                }else {
+                    
+                    if emailText.text != "" && passwordText.text != "" && dateDayCount(date: dateText.text!) > 15 {
+                        
+                        Auth.auth().createUser(withEmail: emailText.text!, password: passwordText.text!) { (authdata,error) in
+                            
+                            if error != nil  {
+                                self.makeAlert(title: "Error!", message: error?.localizedDescription ?? "Error")
+                            }else {
+                                
+                                print("kullanici olustuldu.")
+                            }
+                        }
+                    }
+                }
+            }
+        }
     }
-    */
+    
+    func makeAlert(title:String, message:String) {
+        
+        let alert = UIAlertController(title:title,message:message,preferredStyle: .alert)
+        let okAction = UIAlertAction(title: "Ok", style: .default, handler: nil)
+        
+        alert.addAction(okAction)
+        self.present(alert, animated: true, completion: nil)
+    }
+}
 
+extension String {
+    
+    func isValidEmail(email:String) -> Bool {
+        
+        let emailRegex = "[a-zA-Z0-9.%-]+@[a-zA-Z0-9-]+.[a-zA-Z]{2,4}"
+        let emailTest = NSPredicate(format: "SELF MATCHES %@", emailRegex)
+        let result = emailTest.evaluate(with: email)
+        return result
+    }
+    
+    func isValidPassword(password:String) -> Bool {
+        
+        let passwordRegex = "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{6,10}$"// Minimum 6 maximum 10 characters at least 1 Alphabet and 1 Number
```

Comment:
Regex is not valid for requirement. Password En az 6 karakter olmali ve bir buyuk harf icermeli. A12345 uygun, 123456 uygun degil. (Regex kullanilsin)